controllers/article-controller.js

```diff
@@ -12,7 +12,22 @@ exports.getArticleById = (req, res, next) => {
 }
 
 exports.getAllArticles = (req, res, next) => {
-    fetchAllArticles()
+    let { sort_by, order } = req.query;
+
+    const validSortColumns = ['author', 'title', 'article_id', 'topic', 'created_at', 'votes', 'comment_count']
+    const validOrders = [ 'asc', 'desc']
+        
+    if (sort_by && !validSortColumns.includes(sort_by)){
+      return next({ status: 400, msg: 'Bad Request: Invalid Sort Column'});
+    }
+    if (order && !validOrders.includes(order.toLowerCase())){
+      return next({ status: 400, msg: 'Bad Request: Invalid Sort Order'})
+    }
+
+    sort_by = sort_by || 'created_at';
+    order = order || 'desc';
+
+    fetchAllArticles(sort_by, order)
     .then((articles) => {
         res.status(200).send({ articles });
     })
```

Comment:
Please add a catch block. if any errors are thrown by your model they won't be passed to your middle ware